src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/TokenBasedFormattingRule.cs

```diff
@@ -508,8 +508,13 @@ private static SyntaxList<UsingDirectiveSyntax> GetUsings(SyntaxNode node)
             }
 
             // pointer case for regular pointers
-            if ((currentToken.Kind() == SyntaxKind.AsteriskToken && currentToken.Parent is PointerTypeSyntax) ||
-                (previousToken.Kind() == SyntaxKind.AsteriskToken && previousToken.Parent is PrefixUnaryExpressionSyntax))
+            if (currentToken.Kind() == SyntaxKind.AsteriskToken && currentToken.Parent is PointerTypeSyntax)
+            {
+                return CreateAdjustSpacesOperation(1, AdjustSpacesOption.ForceSpacesIfOnSingleLine);
+            }
+
+            if (previousToken.Kind() == SyntaxKind.AsteriskToken &&
+                (previousToken.Parent is PrefixUnaryExpressionSyntax || previousToken.Kind() == SyntaxKind.IdentifierToken))
```

Comment:
**NOTE**: The original issue complains about NormalizeWhitespace, I'm not sure whether this affects NormalizeWhitespace or not.